samcli/lib/sync/sync_flow_factory.py

```diff
@@ -153,13 +157,33 @@ def _create_lambda_flow(
         resource: Dict[str, Any],
```

Comment:
Can this parameter be removed since this is replaced with the function from the build_context?

Reply:
Unfortunately not, this is called from below (line 373) with generic function signature, if we remove it then it will break that call.

Reply:
Seems like only `_create_lambda_flow` factory method was using that parameter and the other methods had to pass it due to generic signature. Could that parameter be removed from all the methods or would it be useful to keep that parameter for any future resources added?

Reply:
Himm actually you are right and that is a very good catch! Let's see if I can remove it completely.